### PR TITLE
Add DTO's, Automapper

### PR DESCRIPTION
**Pull request by vnu2305**
Added Office repository pattern and DTO

### CODE REVIEW

**Review by vnu2305 — COMMENTED** [review]
review and edits done

File: .gitignore [hunk 1]
```diff
@@ -361,4 +361,7 @@ MigrationBackup/
 .ionide/
 
 # Fody - auto-generated XML schema
-FodyWeavers.xsd
\ No newline at end of file
+FodyWeavers.xsd
```
Comment (svadrif) [inline .gitignore:364]: dont change .gitignore file

File: src/API/API.csproj [hunk 2]
```diff
@@ -18,10 +19,6 @@
     <PackageReference Include="Swashbuckle.AspNetCore" Version="6.2.3" />
   </ItemGroup>
 
-  <ItemGroup>
-    <Folder Include="Controllers\" />
-  </ItemGroup>
```
Comment (svadrif) [inline src/API/API.csproj:23]: you shouldn't delete Controller folder. return it back

File: src/Domain/Domain.csproj [hunk 3]
```diff
@@ -5,4 +5,8 @@
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="System.Text.Json" Version="6.0.2" />
```
Comment (svadrif) [inline src/Domain/Domain.csproj:9]: delete this package

File: src/Domain/Entities/Map.cs [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ﻿using Domain.Common;
 using System;
 using System.Collections.Generic;
+using System.Text.Json.Serialization;
```
Comment (svadrif) [inline src/Domain/Entities/Map.cs:4]: here also

File: src/Domain/Entities/Booking.cs [hunk 5]
```diff
@@ -14,11 +15,9 @@ public class Booking : BaseEntity<Guid>
         /* EF Relation */
```
Comment (svadrif) [inline src/Domain/Entities/Booking.cs:15]: You should change sequence of properties like this

public class Booking : BaseEntity<Guid>
    {
        public DateTime BookingStart { get; set; }
        public DateTime BookingEnd { get; set; }
        public bool IsRecurring { get; set; }
        public string Frequancy { get; set; }
        public bool IsActive { get; set; }
        
        /* EF Relation */
        public Guid UserId { get; set; }
        public AppUser User { get; set; }

        public Guid? ParkingPlaceId { get; set; }
        public ParkingPlace ParkingPlace { get; set; }

        public Guid WorkPlaceId { get; set; }
        public WorkPlace WorkPlace { get; set; }
    }

File: src/Domain/Entities/Map.cs [hunk 6]
```diff
@@ -12,8 +13,7 @@ public class Map : BaseEntity<Guid>
 
         /* EF Relation */
         public IEnumerable<WorkPlace> WorkPlaces { get; set; }
```
Comment (svadrif) [inline src/Domain/Entities/Map.cs:15]: put  public IEnumerable<WorkPlace> WorkPlaces { get; set; } to the end

File: src/Domain/Entities/Booking.cs (hunk 5 above)
Comment (svadrif) [reply src/Domain/Entities/Booking.cs:15]: this comment applies for all entities

File: src/Infrastructure/Services/OfficeService/OfficeService.cs [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        private readonly IMapper _mapper;
+
+        public OfficeService(ApplicationDbContext dataContext,IMapper mapper)
+        {
+            _dataContext = dataContext;
+            _mapper = mapper;
+        }
+        public async Task<ServiceResponse<List<GetOfficeDTO>>>  AddOffice(AddOfficeDTO office)
+        {
+            Office office1 = new Office
```
Comment (svadrif) [inline src/Infrastructure/Services/OfficeService/OfficeService.cs:22]: dont name like this office1
Change AddOfficeDTO office to AddOfficeDTO officeDto
and then change office1 to newOffice for example

File: src/Infrastructure/Services/OfficeService/OfficeService.cs [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+                serviceResponse.Data = (await _dataContext.Offices.ToListAsync()).Select(c=>_mapper.Map<GetOfficeDTO>(c)).ToList();
+                return serviceResponse;
+            }
+                
+            office.Country = office1.Country;
+            office.City = office1.City;
+            office.Address = office1.Address;
+            office.Name = office1.Name;
+            await _dataContext.SaveChangesAsync();
+            var list = await _dataContext.Offices.ToListAsync();
```
Comment (svadrif) [inline src/Infrastructure/Services/OfficeService/OfficeService.cs:62]: give variables meaningful name, like officeList

File: src/Infrastructure/Services/OfficeService/OfficeService.cs [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+            serviceResponse.Data = list.Select(c=>_mapper.Map<GetOfficeDTO>(c)).ToList();
+            return serviceResponse;
+        }
+
+        public async Task<ServiceResponse<List<GetOfficeDTO>>>  GetAllOffices()
+        {
+            var serviceResponse = new ServiceResponse<List<GetOfficeDTO>>();
+            serviceResponse.Data = (await _dataContext.Offices.ToListAsync()).Select(c=>_mapper.Map<GetOfficeDTO>(c)).ToList();
```
Comment (svadrif) [inline src/Infrastructure/Services/OfficeService/OfficeService.cs:42]: firstly assing this  _dataContext.Offices.ToListAsync() to another var and then use this var for mapping

**Review by vnu2305 — COMMENTED** [review]
another review

File: src/Application/Common/AutoMapperProfile.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
+﻿using Application.DTOs.OfficeDTO;
+using AutoMapper;
+using Domain.Entities;
+
+namespace Application.Common;
+
+public class AutoMapperProfile: Profile
```
Comment (svadrif) [inline src/Application/Common/AutoMapperProfile.cs:7]: put it to another folder

File: src/Application/Common/ServiceResponse.cs [hunk 11]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Application.Common;
+
+public class ServiceResponse<T>
+{
+    public T Data { get; set; }
+    public bool success { get; set; } = true;
+    public string message { get; set; } = null;
```
Comment (svadrif) [inline src/Application/Common/ServiceResponse.cs:7]: change string to string? or delete null

File: src/Domain/Entities/Map.cs [hunk 12]
```diff
@@ -11,9 +11,8 @@ public class Map : BaseEntity<Guid>
         public bool HasConfRoom { get; set; }
 
         /* EF Relation */
-        public IEnumerable<WorkPlace> WorkPlaces { get; set; }
-
         public Office Office { get; set; }
```
Comment (svadrif) [inline src/Domain/Entities/Map.cs:14]: here also put OfficeId first and after this put property navigator

File: src/Domain/Entities/ParkingPlace.cs [hunk 13]
```diff
@@ -11,8 +11,7 @@ public class ParkingPlace : IHasKey<Guid>
 
         /* EF Relation */
         public IEnumerable<Booking> Bookings { get; set; }
-
         public Office Office { get; set; }
```
Comment (svadrif) [inline src/Domain/Entities/ParkingPlace.cs:14]: here also put OfficeId first and after this put property navigator

File: src/Domain/Entities/Vacation.cs [hunk 14]
```diff
@@ -13,4 +13,4 @@ public class Vacation : IHasKey<Guid>
         public AppUser User { get; set; }
```
Comment (svadrif) [inline src/Domain/Entities/Vacation.cs:13]: here also put UserId first and after this put property navigator

File: src/Domain/Entities/WorkPlace.cs [hunk 15]
```diff
@@ -19,8 +19,7 @@ public class WorkPlace : BaseEntity<Guid>
 
         /* EF Relation */
         public IEnumerable<Booking> Bookings { get; set; }
-
         public Map Map { get; set; }
```
Comment (svadrif) [inline src/Domain/Entities/WorkPlace.cs:22]: the same here, change MapId and Map places

File: src/Infrastructure/Services/OfficeService/OfficeService.cs [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+    public class OfficeService : IOfficeService
+    {
+
+        private readonly ApplicationDbContext _dataContext;
+        private readonly IMapper _mapper;
+
+        public OfficeService(ApplicationDbContext dataContext,IMapper mapper)
+        {
+            _dataContext = dataContext;
+            _mapper = mapper;
```
Comment (DoniyorJN) [inline src/Infrastructure/Services/OfficeService/OfficeService.cs:19]: please add a null check in a contractor if it's null you should throw an exception
and apply it to every contractor

**Review by DoniyorJN — CHANGES_REQUESTED** [review]

File: src/Infrastructure/Services/OfficeService/OfficeService.cs [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+            var serviceResponse = new ServiceResponse<List<GetOfficeDTO>>();
+            _dataContext.Offices.Add(newOffice);
+            await _dataContext.SaveChangesAsync();
+            var list = await _dataContext.Offices.ToListAsync();
+            serviceResponse.Data = list.Select(c=>_mapper.Map<GetOfficeDTO>(c)).ToList();
+            return serviceResponse;
+        }
+
+        public async Task<ServiceResponse<List<GetOfficeDTO>>>  GetAllOffices()
+        {
```
[the remark this answers is not in the document]
Comment (vnu2305) [reply src/Infrastructure/Services/OfficeService/OfficeService.cs:45]: we might need them for the reports, i guess

File: src/Application/DTOs/MapDTO/GetMapDTO.cs [hunk 18]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Application.DTOs.MapDTO;
+
+public class GetMapDTO
+{
+
```
[the remark this answers is not in the document]
Comment (vnu2305) [reply src/Application/DTOs/MapDTO/GetMapDTO.cs:5]: so, it is better to get rid of them?

File: src/Application/DTOs/MapDTO/AddMapDTO.cs [hunk 19]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Application.DTOs.MapDTO;
+
+public class AddMapDTO
+{
+
```
[the remark this answers is not in the document]
Comment (vnu2305) [reply src/Application/DTOs/MapDTO/AddMapDTO.cs:5]: so, it is better to get rid of them?

File: src/Infrastructure/DependencyInjection.cs [hunk 20]
```diff
@@ -15,7 +15,7 @@ public static void AddInfrastructure(this IServiceCollection services, IConfigur
                     configuration.GetConnectionString("DefaultConnection"),
                     b => b.MigrationsAssembly(typeof(ApplicationDbContext).Assembly.FullName)));
 
-            services.AddScoped<IApplicationDbContext>(provider => provider.GetService<ApplicationDbContext>());
+            services.AddScoped<IApplicationDbContext>(provider => provider.GetService<ApplicationDbContext>() ?? throw new InvalidOperationException());
```
Comment (DoniyorJN) [inline src/Infrastructure/DependencyInjection.cs:18]: why do we need this exception?

**Review by DoniyorJN — CHANGES_REQUESTED** [review]

**Review by svadrif — DISMISSED** [review]

File: .gitignore [hunk 21]
```diff
@@ -1,364 +0,0 @@
-## Ignore Visual Studio temporary files, build results, and
```
Comment (svadrif) [inline .gitignore:1]: Dont delete this file!

Comment (svadrif) [reply .gitignore:1]: Don't confuse .gitignore file with .gitattributes file

**Review by svadrif — APPROVED** [review]

**Pull request merged**